insightvm_connector.py

```diff
@@ -66,193 +68,183 @@ def finalize(self):
+        resp_json = None
 
-        config = self.get_config()
-
-        root = etree.Element(endpoint)
-
-        for k, v in param_dict.items():
-            root.set(k, v)
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)),
+                resp_json
+            )
 
-        if endpoint != 'LoginRequest':
-            root.set('session-id', self._session_id)
+        # Create a URL to connect to
+        url = self._base_url + endpoint
+        self._auth = HTTPBasicAuth(self._username, self._password)
 
         try:
-            response = requests.post(self._base_url, data=ElementTree.tostring(root), headers=self._headers,
-                timeout=consts.INSIGHTVM_DEFAULT_TIMEOUT, verify=config.get(phantom.APP_JSON_VERIFY, False))
+            r = request_func(
+                url,
+                auth=self._auth,
+                json=data,
+                headers=headers,
+                params=params,
+                verify=self._verify,
+            )
         except Exception as e:
-            return RetVal(action_result.set_status(phantom.APP_ERROR, self._process_request_exception(e)), None)
+            self.debug_print("Got exception: {}".format(e))
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(str(e))),
+                resp_json
+            )
 
-        return self._process_response(response, action_result, endpoint)
+        return self._process_response(r, action_result)
 
-    def _response_stripper(self, strippee, ret_dict):
+    def _paginator(self, action_result, endpoint, headers=None, params=None, data=None, method="get", limit=None):
 
-        for k, v in strippee.items():
+        items_list = list()
 
-            k = k.replace('@', '')
-            if '-' in k:
-                k = k.replace('-', ' ')
-                k = k.title()
-            k = k.replace(' ', '')
-            k = k[0].lower() + k[1:]
+        if not params:
+            params = {}
 
-            if type(v) == dict:
-                ret_dict[k] = {}
-                self._response_stripper(v, ret_dict[k])
+        page = 0
+        params["size"] = consts.DEFAULT_MAX_RESULTS
+        params["page"] = page
 
-            elif type(v) == list:
-                ret_dict[k] = []
-                for i in v:
-                    new_dict = {}
-                    self._response_stripper(i, new_dict)
-                    ret_dict[k].append(new_dict)
+        while True:
+            ret_val, items = self._make_rest_call(action_result, endpoint, headers=headers, params=params, data=data, method=method)
 
-            else:
-                ret_dict[k] = v
+            if phantom.is_fail(ret_val):
+                return None
+
+            items_list.extend(items.get("resources"))
```

Comment:
are we implicitly verifying that `resources` exists somewhere? if it's guaranteed to exist we should use `[]` notation since `.get` has the connotation that the key may not exist. otherwise, we should have some verification this isn't ever `None` so we don't have a misleading error due to calling `extend` (or anything else) with `None`

Reply:
Replaced `items_list.extend(items.get("resources"))` with `items_list.extend(items.get("resources", []))` to handle None type exception